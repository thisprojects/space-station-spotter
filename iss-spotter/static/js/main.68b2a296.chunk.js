(this["webpackJsonpspace-station-spotter"]=this["webpackJsonpspace-station-spotter"]||[]).push([[0],{19:function(e,t,n){e.exports=n(43)},24:function(e,t,n){},43:function(e,t,n){"use strict";n.r(t);var a=n(0),r=n.n(a),o=n(5),c=n.n(o),i=(n(24),n(3)),s=n.n(i),l=n(9),u=n(10),d=n(11),p=n(13),m=n(12),f=n(6),h=n(14),v=n(15),g=function(e){return fetch("https://us-central1-nathan-downes-express-api.cloudfunctions.net/api/maps",{method:"POST",body:"".concat(e)}).then((function(e){return e.json()})).then((function(e){return{coords:e.results[0].geometry.location,address:e.results[0].formatted_address}}))},b=function(e,t){return fetch("https://cors-anywhere.herokuapp.com/http://api.open-notify.org/iss-pass.json?lat=".concat(e,"&lon=").concat(t,"&alt=20&n=3")).then((function(e){return e.json()})).then((function(e){return e.response||null}))},E=function(){var e=Object(l.a)(s.a.mark((function e(t){var n,a,r,o,c,i,l;return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:if(n=t,e.t0=!t,!e.t0){e.next=6;break}return e.next=5,new Promise((function(e,t){var n=setTimeout((function(){return t("GeoLocation Time Out")}),1e4);navigator.geolocation.getCurrentPosition((function(t){clearTimeout(n);var a=t.coords.longitude,r=t.coords.latitude;e("".concat(r," ").concat(a))}))}));case 5:n=e.sent;case 6:return e.next=8,g(n);case 8:return a=e.sent,r=a.coords,o=r.lat,c=r.lng,i=a.address,e.next=12,b(o,c);case 12:if(l=e.sent){e.next=15;break}throw new Error("Bad response from space station API");case 15:return e.abrupt("return",{lat:o,lng:c,spaceStationResults:l,address:i});case 16:case"end":return e.stop()}}),e)})));return function(t){return e.apply(this,arguments)}}(),w=n(7),y=n(2),S=function(e){return e<10&&"0".concat(e)||e},O=function(e){return e.results.map((function(e,t){return r.a.createElement("div",{className:"result-row",key:t},r.a.createElement("p",{key:t+1},t+1,r.a.createElement("span",null,r.a.createElement(w.a,{icon:y.a})),function(e){var t=new Date(1e3*e),n=t.getFullYear(),a=["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"][t.getMonth()],r=t.getDate(),o=S(t.getHours()),c=S(t.getMinutes()),i=S(t.getSeconds());return" ".concat(r," / ").concat(a," / ").concat(n,"  at  ").concat(o,":").concat(c,":").concat(i)}(e.risetime)))}))},k=function(e){var t=e.reset;return r.a.createElement("div",{className:"choose-another-location"},r.a.createElement("button",{onClick:function(){return t(!0)}},"Choose Another Location"))},j=function(e){var t=e.results,n=e.reset,a=e.address;return t&&r.a.createElement("div",{className:"results"},r.a.createElement("p",null,a),r.a.createElement(O,{results:t}),r.a.createElement(k,{reset:n}))},C=function(e){return e.error&&r.a.createElement("div",{className:"location-error"},r.a.createElement("p",null,"Error - Location Unavailable Or Unable To Acquire Geolocation"))||null},L=function(e){return e.loading&&r.a.createElement("div",{className:"loading"},r.a.createElement(w.a,{icon:y.c,size:"3x",spin:!0}))},N=function(e){var t=e.submitLocation;return r.a.createElement("div",{className:"geocode"},r.a.createElement("button",{onClick:function(){return t()}},"Use Current Location"))},x=function(e){function t(){var e,n;Object(u.a)(this,t);for(var a=arguments.length,o=new Array(a),c=0;c<a;c++)o[c]=arguments[c];return(n=Object(p.a)(this,(e=Object(m.a)(t)).call.apply(e,[this].concat(o)))).locationRef=r.a.createRef(),n}return Object(h.a)(t,e),Object(d.a)(t,[{key:"render",value:function(){var e=this;return r.a.createElement(r.a.Fragment,null,r.a.createElement("div",{className:"location-input-wrapper"},r.a.createElement("input",{ref:this.locationRef,id:"location-input",type:"text",placeholder:"Choose Location"})),r.a.createElement("div",{className:"submit-button"},r.a.createElement("button",{onClick:function(){return e.props.submitLocation(e.locationRef.current.value)}},"Submit")))}}]),t}(a.Component),A=function(e){var t=e.overlayState,n=e.error,a=e.submitLocation,o=e.loading;return t&&r.a.createElement("div",{className:"overlay"},r.a.createElement("h1",null,"Space Station Spotter"),r.a.createElement("span",{className:"icon"},r.a.createElement(w.a,{icon:y.b,size:"3x"})),r.a.createElement(N,{submitLocation:a}),r.a.createElement("h2",null,"Or"),r.a.createElement("span",{className:"controls"},r.a.createElement(x,{submitLocation:a})),r.a.createElement("p",null,"When can you see the space station? Enter a location to find out!"),r.a.createElement(C,{error:n}),r.a.createElement(L,{loading:o}))||null},R=function(e){function t(e){var n;return Object(u.a)(this,t),(n=Object(p.a)(this,Object(m.a)(t).call(this,e))).state={lat:52.489471,lng:-1.898575,error:!1,spaceStationResults:null,overlayState:!0,loadingState:!1},n.setOverlayState=n.setOverlayState.bind(Object(f.a)(n)),n.setLocation=n.setLocation.bind(Object(f.a)(n)),n}return Object(h.a)(t,e),Object(d.a)(t,[{key:"setOverlayState",value:function(e){this.setState({overlayState:e})}},{key:"setLocation",value:function(){var e=Object(l.a)(s.a.mark((function e(t){var n,a,r,o,c,i,l=this;return s.a.wrap((function(e){for(;;)switch(e.prev=e.next){case 0:return this.setState({loadingState:!0,error:!1}),e.next=3,E(t).catch((function(){l.setState({error:!0,loadingState:!1})}));case 3:n=e.sent,r=(a=n||{}).lat,o=a.lng,c=a.spaceStationResults,i=a.address,!this.state.error&&this.setState({lat:r,lng:o,spaceStationResults:c,address:i,loadingState:!1,overlayState:!1});case 6:case"end":return e.stop()}}),e,this)})));return function(t){return e.apply(this,arguments)}}()},{key:"render",value:function(){var e=this.state,t=e.lat,n=e.lng,a=e.spaceStationResults,o=e.error,c=e.overlayState,i=e.loadingState,s=e.address;return r.a.createElement("div",{id:"page-wrapper"},r.a.createElement(A,{overlayState:c,submitLocation:this.setLocation,error:o,loading:i}),r.a.createElement("div",{id:"map",className:"map"},r.a.createElement(v.Map,{initialCenter:{lat:52.489471,lng:-1.898575},center:{lat:t,lng:n},google:this.props.google,zoom:3,className:"map"},r.a.createElement(v.Marker,{position:{lat:t,lng:n},name:"Current location"})),r.a.createElement(j,{reset:this.setOverlayState,results:a,address:s})))}}]),t}(a.Component),W=Object(v.GoogleApiWrapper)({apiKey:"AIzaSyCyNv5BOZZfdKO3VDhQCOA3Ufm8tv8rCF8"})(R),T=Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));function U(e,t){navigator.serviceWorker.register(e).then((function(e){e.onupdatefound=function(){var n=e.installing;null!=n&&(n.onstatechange=function(){"installed"===n.state&&(navigator.serviceWorker.controller?(console.log("New content is available and will be used when all tabs for this page are closed. See https://bit.ly/CRA-PWA."),t&&t.onUpdate&&t.onUpdate(e)):(console.log("Content is cached for offline use."),t&&t.onSuccess&&t.onSuccess(e)))})}})).catch((function(e){console.error("Error during service worker registration:",e)}))}c.a.render(r.a.createElement(W,null),document.getElementById("root")),function(e){if("serviceWorker"in navigator){if(new URL(".",window.location.href).origin!==window.location.origin)return;window.addEventListener("load",(function(){var t="".concat(".","/service-worker.js");T?(!function(e,t){fetch(e,{headers:{"Service-Worker":"script"}}).then((function(n){var a=n.headers.get("content-type");404===n.status||null!=a&&-1===a.indexOf("javascript")?navigator.serviceWorker.ready.then((function(e){e.unregister().then((function(){window.location.reload()}))})):U(e,t)})).catch((function(){console.log("No internet connection found. App is running in offline mode.")}))}(t,e),navigator.serviceWorker.ready.then((function(){console.log("This web app is being served cache-first by a service worker. To learn more, visit https://bit.ly/CRA-PWA")}))):U(t,e)}))}}()}},[[19,1,2]]]);
//# sourceMappingURL=main.68b2a296.chunk.js.map