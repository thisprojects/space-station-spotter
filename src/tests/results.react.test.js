import React from "react";
import DisplayResults, {
  MapResults,
  ChooseAnotherLocation,
  unixTimeConverter
} from "../components/results";
import renderer from "react-test-renderer";
import { mockedResultsArray } from "./mocked-results.js";

test("Unix Time Conversion", () => {
  let x = unixTimeConverter(1580051829);
  expect(x).toEqual(" 26 / Jan / 2020  at  15:17:09");
});

test("MapResults Matches Snapshot", () => {
  const component = renderer.create(
    <MapResults results={mockedResultsArray} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("DisplayResults Matches Snapshot", () => {
  const component = renderer.create(
    <DisplayResults results={mockedResultsArray} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Choose Another Location Matches Snapshot", () => {
  const component = renderer.create(<ChooseAnotherLocation />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
