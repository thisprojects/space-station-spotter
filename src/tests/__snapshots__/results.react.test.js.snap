// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choose Another Location Matches Snapshot 1`] = `
<div
  className="choose-another-location"
>
  <button
    onClick={[Function]}
  >
    Choose Another Location
  </button>
</div>
`;

exports[`DisplayResults Matches Snapshot 1`] = `
<div
  className="results"
>
  <p />
  <div
    className="result-row"
  >
    <p>
      1
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:03:08
    </p>
  </div>
  <div
    className="result-row"
  >
    <p>
      2
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:13:02
    </p>
  </div>
  <div
    className="result-row"
  >
    <p>
      3
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:08:57
    </p>
  </div>
  <div
    className="choose-another-location"
  >
    <button
      onClick={[Function]}
    >
      Choose Another Location
    </button>
  </div>
</div>
`;

exports[`MapResults Matches Snapshot 1`] = `
Array [
  <div
    className="result-row"
  >
    <p>
      1
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:03:08
    </p>
  </div>,
  <div
    className="result-row"
  >
    <p>
      2
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:13:02
    </p>
  </div>,
  <div
    className="result-row"
  >
    <p>
      3
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-clock fa-w-16 "
          data-icon="clock"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </span>
       27 / Jan / 2020  at  15:08:57
    </p>
  </div>,
]
`;
